### PR TITLE
test: #132 rp adapter: bdd test for didexchange with wallet

### LINKED ISSUE #132: rp adapter: bdd test - didexchange with wallet
part of #66

### PR DESCRIPTION
**Pull request by llorllale**
closes #132

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: test/bdd/pkg/rp/rp_steps.go [hunk 1]
```diff
@@ -10,56 +10,80 @@ import (
...
 	g.Step(`^the client ID of the tenant with label "([^"]*)" is registered at hydra$`, s.lookupClientID)
+	g.Step(`^a registered rp tenant with label "([^"]*)" and callback "([^"]*)"$`, s.registerTenantFlow)
+	g.Step(`^the rp tenant "([^"]*)" redirects the user to the rp adapter$`, s.redirectUserToAdapter)
+	g.Step(`the rp adapter "([^"]*)" submits a CHAPI request to "([^"]*)" with presentation-definitions and a didcomm invitation to connect`, s.sendCHAPIRequestToWallet) //nolint:lll
+	g.Step(`^"([^"]*)" accepts the didcomm invitation$`, s.walletAcceptsDIDCommInvitation)
+	g.Step(`^"([^"]*)" connects with the RP adapter "([^"]*)"$`, s.validateConnection)
+}
+
+func (s *Steps) registerAgentController(agentID, inboundHost, inboundPort, controllerURL string) error {
+	return s.controller.ValidateAgentConnection(agentID, inboundHost, inboundPort, controllerURL)
```
Comment (rolsonquadras) [inline test/bdd/pkg/rp/rp_steps.go:86]: Can't we use the step definition from agent_controller? This just looks to be passthrough with no extra logic.

File: test/bdd/pkg/rp/rp_steps.go [hunk 2]
```diff
@@ -10,56 +10,80 @@ import (
...
 func (s *Steps) RegisterSteps(g *godog.Suite) {
+	g.Step(`^the "([^"]*)" is running on "([^"]*)" port "([^"]*)" with controller "([^"]*)"$`, s.registerAgentController)
 	g.Step(`^a request is sent to create an RP tenant with label "([^"]*)" and callback "([^"]*)"$`, s.createTenant)
 	g.Step(`^the trustbloc DID of the tenant with label "([^"]*)" is resolvable$`, s.resolveDID)
 	g.Step(`^the client ID of the tenant with label "([^"]*)" is registered at hydra$`, s.lookupClientID)
+	g.Step(`^a registered rp tenant with label "([^"]*)" and callback "([^"]*)"$`, s.registerTenantFlow)
+	g.Step(`^the rp tenant "([^"]*)" redirects the user to the rp adapter$`, s.redirectUserToAdapter)
+	g.Step(`the rp adapter "([^"]*)" submits a CHAPI request to "([^"]*)" with presentation-definitions and a didcomm invitation to connect`, s.sendCHAPIRequestToWallet) //nolint:lll
+	g.Step(`^"([^"]*)" accepts the didcomm invitation$`, s.walletAcceptsDIDCommInvitation)
+	g.Step(`^"([^"]*)" connects with the RP adapter "([^"]*)"$`, s.validateConnection)
```
Comment (rolsonquadras) [inline test/bdd/pkg/rp/rp_steps.go:82]: Minor observation : It's better to separate wallet steps definition from RP definitions. For Issuer adapter, I tried to add wallet related things in agent_controller_steps and issuer actions in issuer_steps.

Comment (llorllale) [reply test/bdd/pkg/rp/rp_steps.go:82]: I think it's better to leave each BDD test to define its own steps. I think the "agent steps" should really be an SDK/utility that can be used by the actual BDD test files.

File: test/bdd/pkg/rp/rp_steps.go (hunk 1 above)
Comment (llorllale) [reply test/bdd/pkg/rp/rp_steps.go:86]: It's a passthrough unto a different, internal instance. These tests are not interested in registering this wallet agent in the other, "global" instance of the agent controller steps.

**Pull request merged**